### ISSUE TITLE
Meetings Notes

### ISSUE THREAD

**Issue by kaiyun717** [kaiyun717/cs285_project#7]
## 10/05/2022
- Codebase: [https://github.com/tung-nd/E2C-pytorch](https://github.com/tung-nd/E2C-pytorch)
    - Kyle has made it running and, in theory, training.
    - Change `compute_loss` to make it not binary.

**Comment by kaiyun717** [comment]
## 10/12/2022
---

- Train E2C in couple of different environments.
    ```python
    torch.autograd.functional.jvp(dynamics, (x, u), (dx, du)) -> dx(t+1)
    ```

- Include $$KL(\delta z(t+1) || \hat{\delta z}(t+1))$$ to loss function. If the performance with this updated loss function is better than the E2C, then the Jacobian is not a "perfect" linearization: `consis_term`. Let's call this the "Jacobian term". If the errors are very small, there's not much reason to add it to the loss function. @kylestach 
    - Experiment with MSE in the future. Right now, stick to KL-divergence.
    - Reach: Experiment with Hopper.

- Nonlinear E2C: "linearize during planning" -> our guess is that they are utilizing finite differences to estimate the Jacobians during planning. Or autograd.

- Pull in Hopper (MuJoCo) to the environment. @kaiyun717 @Haoming-Guo.

- Run experiments once Colab is set @all.

**Comment by kaiyun717** [comment]
## 10/22/2022
---
- Implement Hopper Sampler @kaiyun717 
    - **Images**: 64 x 64 frame (further 128 x 128)
        - Variable frame concat size
        - Get rid of `shards`
    - **Obs**: Pandas Dataframe instead of JSON file

- Planar (z-dim: 8) @Haoming-Guo 

- CNN implementation for `networks.py` @Haoming-Guo 

- Jacobian @kylestach
